rsocket-transport-ktor/rsocket-transport-ktor-server/src/jvmMain/kotlin/io/rsocket/kotlin/core/RSocketServerBuilders.kt

```diff
@@ -17,21 +17,21 @@
 package io.rsocket.kotlin.core
 
 import io.ktor.application.*
-import io.ktor.http.*
 import io.ktor.routing.*
 import io.ktor.websocket.*
 import io.rsocket.kotlin.*
 import io.rsocket.kotlin.connection.*
 
 fun Route.rSocket(path: String, protocol: String? = null, acceptor: RSocketAcceptor) {
-    route(path, HttpMethod.Get) {
-        rSocket(protocol, acceptor)
+    val feature = application.feature(RSocketServerSupport)
+    webSocket(path = path, protocol = protocol) {
+        connection.startServer(feature.configuration, acceptor).join()
     }
 }
 
 fun Route.rSocket(protocol: String? = null, acceptor: RSocketAcceptor) {
```

Comment:
why both instead of null as a default above?

Reply:
for parity with ktor routes - they has the same structure because, if path is provided, underhood it creates new `route` object. We can use `path: String? = null` and then we will need to add `if` inside, to do appropriate call